src/sagas/FetchSaga.test.ts

```diff
@@ -0,0 +1,146 @@
+import { takeEvery, call, select } from "redux-saga/effects";
+import { describe } from 'riteway'
+
+import { Actions } from '../actions'
+import { fetchSaga, fetchData, dispatchFetchStatusUpdate, getLatestTwoNamesOnResource, apiFetch, TEXT, NOT_FOUND } from './FetchSaga'
+import { FetchType } from '../reducers/FetchReducer'
+import { getResourceState } from '../selectors/fetch'
+
+describe('FetchSaga()', async (should: any) => {
+  const { assert } = should()
+
+  const iterator = fetchSaga()();
+
+  assert({
+    given: 'fetchRequest Action',
+    should: 'handle fetch request',
+    actual: iterator.next().value,
+    expected: takeEvery(Actions.fetchRequest, fetchData),
+  })
+})
+
+describe('fetchData() Not Found Error', async (should: any) => {
+  const { assert } = should()
+  const url = { url: 'test'}
+  const short = getLatestTwoNamesOnResource(url.url)
```

Comment:
Deconstructing a 'url' variable would remove the, slightly awkward, use of 'url.url':
```javascript
const { url } = { url: 'test'}
const short = getLatestTwoNamesOnResource(url)
```